packages/server/api/controllers/movies.controller.js

```diff
@@ -25,6 +25,47 @@ const getMovieByID = async (id) => {
   }
 };
 
+const getDetailsOfMovieByID = async (id) => {
+  if (!id) {
+    throw new HttpError('ID should be a number', 400);
+  }
+
+  try {
+    const movie = await knex
+      .select(
+        'm.id',
+        'm.title',
+        'm.description',
+        'm.movie_year',
+        'm.image_location',
+        knex.raw('ROUND(AVG(r.rating),1) as rating'),
+        'c.name as category_name',
+        knex.raw(
+          'GROUP_CONCAT(CASE WHEN mc.role = "Director" THEN cm.full_name END) as director',
+        ),
+        knex.raw(
+          'GROUP_CONCAT(CASE WHEN mc.role = "Writer" THEN cm.full_name END) as writer',
+        ),
+      )
+      .from('movies as m')
+      .leftJoin('reviews as r', 'm.id', 'r.movie_id')
```

Comment:
Are these leftjoins necessary to provide some information of moview review, categories etc?

Reply:
i think using leftjoins returns the movie record even if there is no related data in the other tables, some movie have no categories so this will return null. is this correct?

Reply:
Okay so you movie details must contain categories also?
if that is the case then its true but have you verified if api returns categories?

Reply:
if the movie have no categories assigned to it it will return null like this example here
```
{
    "id": 1,
    "title": "The Matrix",
    "description": "When a beautiful stranger leads computer hacker Neo to a forbidding underworld, he discovers the shocking truth--the life he knows is the elaborate deception of an evil cyber-intelligence.",
    "movie_year": 1999,
    "image_location": null,
    "rating": 4,
    "number_of_ratings": 3,
    "category_name": null,
    "director": null,
    "writer": null
}
```
Otherwise it will return it like this example here 
```
{
    "id": 9,
    "title": "Stuart Little",
    "description": "The adventures of a heroic and debonair stalwart mouse named Stuart Little with human qualities, who faces some comic misadventures while searching for his lost bird friend and living with a human family as their child.",
    "movie_year": 1999,
    "image_location": "https://www.themoviedb.org/t/p/w1280/jwydlxxLPVxM5uCfJybEsANzvvg.jpg",
    "rating": null,
    "number_of_ratings": 0,
    "category_name": "Fantasy",
    "director": "Rob Minkoff",
    "writer": null
}
```